# TensorFlow Learning Part 1: Environment  
I have created a new environment (tf_current) that has the latest version of TensorFlow. I will keep my (root) environment the way it is since I have a few talks planned and I don't want to break anything.

**Creating the Environment**  
Here are the commands I created using Anaconda on Windows 10  

Create the environment using Python 3.6:  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;conda create -n tf_current python=3.6  
Change to the new environment:  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;activate tf_current  
Install the latest TensorFlow:  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pip install tensorflow  
Install the module that will allow me to select a different environment within the notebooks:  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;conda install nb_conda  
Install Pandas for some math work  
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;pip install pandas

In [4]:
import tensorflow as tf

In [6]:
#Validate Version
print (tf.__version__)

1.12.0


**Generic Classification**

In [11]:
mnist = tf.keras.datasets.mnist

(x_train,y_train),(x_test,y_test) = mnist.load_data() #Load the data and split into Train/Test
x_train, x_text = x_train / 255.0, x_test / 255.0

In [14]:
#Create the model
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(512, activation=tf.nn.relu),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])
model.compile(optimizer='adam',
             loss='sparse_categorical_crossentropy',
             metrics=['accuracy']) #Returns accuracy from 'evaluate'

model.fit(x_train,y_train,epochs=5)
model.evaluate(x_test,y_test) #Returns [Loss,Accuracy]

Epoch 1/5
60000/60000 [==============================] - 12s 202us/step - loss: 0.2005 - acc: 0.9411
Epoch 2/5
60000/60000 [==============================] - 12s 193us/step - loss: 0.0806 - acc: 0.9756
Epoch 3/5
60000/60000 [==============================] - 12s 192us/step - loss: 0.0518 - acc: 0.9833
Epoch 4/5
60000/60000 [==============================] - 11s 191us/step - loss: 0.0365 - acc: 0.9883
Epoch 5/5
10000/10000 [==============================] - 0s 42us/step


[0.34913151187835684, 0.9782]

As you can see with the results we had a loss of 0.35 and an accuracy score of 0.9782.